tools/runner-setup.sh

```diff
@@ -103,3 +106,27 @@ if [ "$visionOS" = "true" ]; then
         echo_succ "Found some visionOS Simulator runtime supporting OS '$os'. Skipping..."
     fi
 fi
+
+if [ "$ssh" = "true" ]; then
+    # Adds SSH config, so we can git clone GH repos.
+    echo_subtitle "Add SSH configuration"
+    SSH_KEY_PATH="$HOME/.ssh/id_ed25519"
+    SSH_CONFIG_PATH="$HOME/.ssh/config"
+
+    if [ ! -f "$SSH_KEY_PATH" ] || [ ! -f "$SSH_CONFIG_PATH" ]; then
+        echo_warn "Found no SSH key or SSH config file. Configuring..."
+        get_secret $DD_IOS_SECRET__SSH_KEY > $SSH_KEY_PATH
+        chmod 600 "$SSH_KEY_PATH"
+
+        cat <<EOF > "$HOME/.ssh/config"
+Host github.com
+    HostName github.com
+    User git
+    IdentityFile $SSH_KEY_PATH
+    StrictHostKeyChecking no
+EOF
```

Comment:
I imagine we need this to clone the repo but per my comment above, we shouldn't need to clone it again and just reset the state with some git commands.

Reply:
Sure 👌 (answered in other comment). However, the `--ssh` setup for runner will be also utilized for [SR Snapshot Tests](https://datadoghq.atlassian.net/wiki/x/nIxdu) and dogfooding automations that I will be migrating next (both do git clone). It is also used for XCFramework Smoke Tests in this PR.